src/pages/Swap.tsx

```diff
@@ -232,10 +234,10 @@ function Swap(): ReactElement {
       }
       const amountToGive = parseUnits(
         cleanedFormFromValue,
-        TOKENS_MAP[formStateArg.from.symbol].decimals,
+        tokensMap?.[formStateArg.from.symbol]?.decimals ?? 0,
       )
-      const tokenFrom = TOKENS_MAP[formStateArg.from.symbol]
-      const tokenTo = TOKENS_MAP[formStateArg.to.symbol]
+      const tokenFrom = tokensMap[formStateArg.from.symbol]
```

Comment:
hoist both of these to the top of the function. I think we should either assume that if these tokens are in `formStateArg`, then they must also be in `tokensMap`, then we can force them to be `BasicToken` and not undefined. Or we check if they're undefined around line 213 and not even try to use them